test/jdk/java/net/httpclient/HttpResponseInputStreamInterruptTest.java

```
@@ -0,0 +1,172 @@
+                os.write(workingResponse);
+                os.flush();
+
+                // await the interrupt threads completion, then write the second message
+                interruptReadyLatch.countDown();
+                interruptDoneLatch.await();
+                os.write(errorResponse);
+            } catch (InterruptedException e) {
+                System.out.println("interruptDoneLatch await failed");
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    static Thread createClientThread(CountDownLatch interruptReadyLatch, int port) {
+        return new Thread(() -> {
+            try {
+                HttpClient client = HttpClient
+                        .newBuilder()
+                        .proxy(HttpClient.Builder.NO_PROXY)
+                        .build();
+
+                URI uri = URIBuilder.newBuilder()
+                        .scheme("http")
+                        .loopback()
+                        .port(port)
+                        .path("/HttpResponseInputStreamInterruptTest/")
+                        .build();
+
+                HttpRequest request = HttpRequest
+                        .newBuilder(uri)
+                        .GET()
+                        .build();
+
+                // Send a httpRequest and assert the first response is received as expected
+                HttpResponse<InputStream> response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
+                String firstOutput = new String(response.body().readNBytes(FIRST_MESSAGE.getBytes().length));
+                assertEquals(firstOutput, FIRST_MESSAGE);
+
+                // countdown on latch, and assert that an IOException is throw due to the interrupt
+                interruptReadyLatch.countDown();
+                assertThrows(IOException.class, () -> response.body().readAllBytes(), "expected IOException");
+            } catch (Exception e) {
+                throw new RuntimeException(e);
```

You might want to catch Throwable, then print the stack trace and call Asserts.fail here - because unhandled exceptions in threads might just be swallowed.